core/cloudflow-operator/src/main/scala/cloudflow/operator/event/ConfigInputChangeEvent.scala

```diff
@@ -0,0 +1,259 @@
+                labels =
+                  CloudflowLabels(app).baseLabels ++ Map(
+                        Operator.AppIdLabel          -> appId,
+                        Operator.StreamletNameLabel  -> streamletName,
+                        CloudflowLabels.ConfigFormat -> CloudflowLabels.OutputConfig
+                      ),
+                ownerReferences = CloudflowApplication.getOwnerReferences(app)
+              ),
+              data = Map(
+                ConfigInputChangeEvent.SecretDataKey -> streamletConfig
+                      .root()
+                      .render(ConfigRenderOptions.concise())
+                      .getBytes(StandardCharsets.UTF_8)
+              )
+            )
+            Action.create(outputSecret, secretEditor)
+          }
+
+        case _ ⇒ Nil // app could not be found, do nothing.
+      }
+      .mapConcat(_.toList)
+
+  def getConfigFromSecret(configInputChangeEvent: ConfigInputChangeEvent, system: ActorSystem) = {
+    val secret: Secret = configInputChangeEvent.watchEvent._object
+    secret.data
+      .get(ConfigInputChangeEvent.SecretDataKey)
+      .map { bytes =>
+        Try(ConfigFactory.parseString(new String(bytes, StandardCharsets.UTF_8))).toOption.getOrElse {
+          system.log.error(
+            s"Detected input secret '${secret.metadata.name}' contains invalid configuration data, IGNORING configuration."
+          )
+          ConfigFactory.empty()
+        }
+      }
+      .getOrElse {
+        system.log.error(
+          s"Detected input secret '${secret.metadata.name}' does not have data key '${ConfigInputChangeEvent.SecretDataKey}', IGNORING configuration."
+        )
+        ConfigFactory.empty()
+      }
+  }
+
+  def streamletConfigPath(streamletName: String)        = s"cloudflow.streamlets.$streamletName"
+  def streamletRuntimeConfigPath(streamletName: String) = s"cloudflow.streamlets.$streamletName.config"
+  def globalRuntimeConfigPath(runtime: String)          = s"cloudflow.runtimes.$runtime.config"
+
+  def getStreamletConfig(streamletName: String, appConfig: Config, runtimeConfig: Config) = {
+    val path = streamletConfigPath(streamletName)
+
+    Try(appConfig.getConfig(path)).toOption
+      .getOrElse(ConfigFactory.empty())
+      .atPath(path)
+      .withFallback(runtimeConfig)
+  }
+
+  def getStreamletRuntimeConfig(runtime: String, streamletName: String, appConfig: Config) =
+    Try(appConfig.getConfig(globalRuntimeConfigPath(runtime))).toOption
+      .getOrElse(ConfigFactory.empty())
+      .atPath(streamletRuntimeConfigPath(streamletName))
+
+  def moveConfigParameters(config: Config, streamletName: String): Config = {
+    val key                        = streamletConfigPath(streamletName)
+    val configParametersKey        = "config-parameters"
+    val absoluteConfigParameterKey = s"$key.$configParametersKey"
+    val configParametersSection = Try(Some(config.getConfig(absoluteConfigParameterKey)))
```

Comment:
`Try(config.getConfig(absoluteConfigParameterKey)).toOption` ?

Reply:
Weird. yes!

Reply:
(I use that everywhere.. brain glitch)